wasm-text.md

```diff
@@ -744,16 +775,22 @@ They distribute the text-to-abstract functions above over lists.
 
     rule #t2aInstrs<C>(I:Instr IS:Instrs) => #t2aInstr<C>(I) #t2aInstrs<C>(IS)
     rule #t2aInstrs<_>(.Instrs) => .Instrs
-
-    rule #t2aLocalDecls<C>(LD:LocalDecl LDS:LocalDecls) => #t2aLocalDecl<C>(LD) #t2aLocalDecls<C>(LDS)
-    rule #t2aLocalDecls<_>(.LocalDecls) => .LocalDecls
 ```
 
 ### Functions for Gathering Context
 
 The following are helper functions for gathering and updating context.
 
 ```k
+    syntax Map ::= #idcTypes    ( Defns      ) [function]
+                 | #idcTypesAux ( Defns, Int ) [function]
+ // -----------------------------------------------------
+    rule #idcTypes(DEFNS) => #idcTypesAux(DEFNS, 0)
+
+    rule #idcTypesAux((type ID:Identifier (func _)) TS, IDX) => (ID |-> IDX) #idcTypesAux(TS, IDX +Int 1)
+    rule #idcTypesAux((type               (func _)) TS, IDX) =>              #idcTypesAux(TS, IDX +Int 1)
+    rule #idcTypesAux(.Defns, _) => .Map
```

Comment:
Please use map update instead of map concatenation, and consider making tail recursive if the resulting map will be very large.

```suggestion
    rule #idcTypesAux((type ID:Identifier (func _)) TS, IDX) => #idcTypesAux(TS, IDX +Int 1) [ ID <- IDX ]
    rule #idcTypesAux((type               (func _)) TS, IDX) =>              #idcTypesAux(TS, IDX +Int 1)
    rule #idcTypesAux(.Defns, _) => .Map
```

Reply:
Map concat fails at runtime ungracefully if the key is already defined. Map update does the correct thing.

Reply:
Okay. I can add a side condition, too, saying the key is not already present (because duplicate IDs are not allowed).

Reply:
Done. Using map update (with a side condition) and tail recursion.